cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/config/python/PythonStdLibConfigurationPass.kt

```diff
@@ -26,116 +26,182 @@
+     */
+    private fun handleConstructExpression(expr: ConstructExpression): Configuration? {
+        if (expr.name.toString() == "configparser.ConfigParser") {
+            val conf = Configuration(underlyingNode = expr)
+            expr.prevDFG += conf
+            return conf
+        }
 
-            if (filename == null) {
-                log.warn("Could not determine filename for configparser.ConfigParser.read")
-                return null
-            }
+        return null
+    }
 
-            // Look for config nodes that match the file name
-            val conf =
-                tu.translationResult?.conceptNodes?.filterIsInstance<Configuration>()?.filter {
-                    it.underlyingNode?.name.toString() == filename
-                }
-            return conf?.map {
-                // Create a LoadConfigurationFile operation for each matching config node
-                val op = LoadConfigurationFile(call, it)
-
-                // And add a dataflow edge into the config node. This is a bit trickier, we want
-                // to find the construct expression to the ConfigParser object and then inject it.
-                // This is not perfect but the read() operation is applied on the object itself, and
-                // we would need to set this point as the last write, which we cannot do here.
-                val test = call.base?.followPrevDFG { it is ConstructExpression }
-                val construct = test?.lastOrNull()
-                if (construct is ConstructExpression) {
-                    construct.prevDFGEdges.addContextSensitive(
-                        it,
-                        callingContext = CallingContextOut(construct),
-                    )
+    /**
+     * Translates `configparser.ConfigParser.read(filename)` calls into [LoadConfiguration]
+     * operations.
+     */
+    private fun handleMemberCallExpression(call: MemberCallExpression): List<LoadConfiguration>? {
+        val firstArgument = call.arguments.firstOrNull()
+        if (call.name.toString() == "configparser.ConfigParser.read" && firstArgument != null) {
+            // Look for our config data structure based on our "base" object
+            val paths =
+                call.base?.followDFGEdgesUntilHit(direction = Backward(GraphToFollow.DFG)) {
+                    it is Configuration
                 }
+            paths?.fulfilled?.map {
+                val conf = it.last() as Configuration
+                val op = LoadConfiguration(call, conf, fileExpression = firstArgument)
                 op
             }
         }
 
         return null
     }
 
+    /**
+     * Translates `config["group"]` and `config["group"]["option"]` into operations, such as
+     * [ReadConfigurationGroup] or [ReadConfigurationOption].
+     *
+     * Since the `configparser` module does not provide a way to explicitly define/register options
+     * or groups, we need to implicitly create them here as well.,
```

Comment:
With "explicitly define/register" you don't mean `add_section(section)` or `set(section, option, value)` from `ConfigParser`, do you?